synapse/models/files.py

```diff
@@ -183,8 +183,9 @@ async def initCoreModule(self):
     async def _hookFileBytesSha256(self, node, prop, norm):
         # this gets called post-norm and curv checks
         if node.ndef[1].startswith('sha256:'):
```

Comment:
It seems like this check is redundant (and maybe slower? Is `.startswith()` less performant than the f-string and comparison?). You could probably remove this line and outdent the three lines you added.

Reply:
There are instances where the string starts with `guid:` that we don't want to check for matching the norm'd fstring :+1: